crates/nu-cli/tests/completions/mod.rs

```diff
@@ -2944,3 +2911,37 @@ fn type_inferenced_operator_completions(mut custom_completer: NuCompleter) {
     let expected: Vec<_> = vec!["=", "=="];
     match_suggestions(&expected, &suggestions);
 }
+
+#[test]
+fn flag_filepath_completions() {
+    use support::completions_helpers::new_engine_helper;
+    use tempfile::tempdir;
+
+    let test_dir = tempdir().unwrap();
+    let test_dir_path = test_dir.path();
+
+    std::fs::write(test_dir_path.join("socket1.sock"), "").unwrap();
+    std::fs::write(test_dir_path.join("socket2.sock"), "").unwrap();
+    std::fs::create_dir(test_dir_path.join("subdir")).unwrap();
+
+    let test_dir_abs = AbsolutePathBuf::try_from(test_dir_path).unwrap();
+    let (_dir, _dir_str, engine, stack) = new_engine_helper(test_dir_abs);
+    let mut completer = NuCompleter::new(Arc::new(engine), Arc::new(stack));
+
+    // Long flag
+    let suggestions = completer.complete("http get --unix-socket ./s", 26);
```

Comment:
Instead of `http get`, you can define a dummy command just for this test to separate the `http get` changes from the completer changes